app/assets/javascripts/backbone/views/project_step_modal_view.coffee

```diff
@@ -75,9 +77,55 @@ class MS.Views.ProjectStepModalView extends Backbone.View
     @done()
     @done = (->) # Reset to empty function.
 
+  setScheduledEndDate: ->
+    # Applies to all steps
+    # Takes the current start date plus current duration and changes the end date
+
+    startDateVal = @$('#project_step_scheduled_start_date').val()
+    startDate = new Date(startDateVal)
+
+    durationVal = @$('#project_step_scheduled_duration_days').val()
+    duration = Number(durationVal)
+
+    endDate = new Date(startDateVal)
+    endDate.setDate(endDate.getDate() + duration)
+    endDateMoment = moment(endDate)
+    endDateFormatted = moment(endDateMoment).format(MS.dateFormats.default)
+
+    @$(".form-group.project_step_scheduled_end_date").find(".static-text-as-field").
+      html(endDateFormatted)
+
+  setScheduledStartDateOnDependent: ->
+    # Applies to dependent step only
+    # Set start date to the precedent step end date plus 1
+    precedentId = @$('#project_step_schedule_parent_id').val()
+    if precedentId
+      startDate = $(".step-end-date[data-id=#{precedentId}]").data('dependent-step-start-date')
```

Comment:
If we handed of an object with all of the steps in the timeline in this format, maybe we could get around this. Yes, there is not time in this issue though.

precedentStepId: dependentStartDate

{45: "May 6, 2017", 67: "July 9, 2017"}, etc.